File: ce/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples43.vb
' <snippetlinqexamples43>


 Using svcContext As New ServiceContext(_serviceProxy)
  Dim query_pagingsort3 = ( _
      From c In svcContext.ContactSet _
      Where c.LastName.StartsWith("W") _
      Order By c.MiddleName Ascending, c.FirstName Descending _
      Select New With {Key c.FirstName,
                       Key c.MiddleName,
                       Key c.LastName}).Take(10)
  For Each c In query_pagingsort3
   Console.WriteLine(c.FirstName &amp; " " &amp; c.MiddleName &amp; " " &amp; c.LastName)
  Next c
 End Using

' </snippetlinqexamples43>

File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples43.cs
// <snippetlinqexamples43>


 using (ServiceContext svcContext = new ServiceContext(_serviceProxy))
 {
  var query_pagingsort3 = (from c in svcContext.ContactSet
                           where c.LastName.StartsWith("W")
                           orderby c.MiddleName ascending,
                           c.FirstName descending
                           select new
                           {
                            c.FirstName,
                            c.MiddleName,
                            c.LastName
                           }).Take(10);
  foreach (var c in query_pagingsort3)
  {
   System.Console.WriteLine(c.FirstName + " " +
    c.MiddleName + " " + c.LastName);
  }
 }

// </snippetlinqexamples43>